Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -309,27 +296,23 @@ def checkable_urls(formula)
     # Preprocesses and returns the URL used by livecheck.
     # @return [String]
     def preprocess_url(url)
-      # Check for GitHub repos on github.com, not AWS
-      url = url.sub("github.s3.amazonaws.com", "github.com") if url.include?("github")
-
-      # Use repo from GitHub or GitLab inferred from download URL
-      if url.include?("github.com") && GITHUB_SPECIAL_CASES.none? { |sc| url.include? sc }
-        if url.include? "archive"
-          url = url.sub(%r{/archive/.*}, ".git") if url.include? "github"
-        elsif url.include? "releases"
-          url = url.sub(%r{/releases/.*}, ".git")
-        elsif url.include? "downloads"
-          url = "#{Pathname.new(url.sub(%r{/downloads(.*)}, "\\1")).dirname}.git"
-        elsif !url.end_with?(".git")
-          # Truncate the URL at the user/repo part, if possible
-          %r{(?<github_repo_url>(?:[a-z]+://)?github.com/[^/]+/[^/#]+)} =~ url
-          url = github_repo_url if github_repo_url.present?
-
-          url.delete_suffix!("/") if url.end_with?("/")
-          url += ".git"
-        end
-      elsif url.include?("/-/archive/")
-        url = url.sub(%r{/-/archive/.*$}i, ".git")
+      uri = URI.parse url
```

Comment:
> As mentioned in the review comment, we need to handle GitHub's older URL format like `https://github.com/downloads/ryankee/growly/growly-v0.2.0.tar.gz`

TIL `github.com/downloads/<owner>/<repo>` is actually just a _proxy_ to `github.s3.amazonaws.com/<owner>/<repo>`. 

> Doing the substitution at the start of the method allows us to handle both of these cases using the same code, instead of duplicating it in an `if` block (as is being done below).

This is smart, I like it 👍  Although checking with `.include?` seems risky, how about staying on the safer side with:
```rb
host = url.host === "github.s3.amazonaws.com" ? "github.com" : url.host
```
and yeah I'm fully aware that chances of hitting URL **pathname** including `github.s3.amazonaws.com` are almost nonexistent 🤷 

> [For what it's worth, I'm working on updating these older GitHub URLs in formulae to the current format but I think it's still good to handle them, especially since the same code can be leveraged for the `github.s3.amazonaws.com` URLs.]

Aren't we suppose to go in another direction? 🤔  `github.s3.amazonaws.com` seems like Github's implementation detail _hidden_ behind the `github.com/downloads` public _contract_.